internal/controller/datadoggenericresource/finalizer_test.go

```diff
@@ -65,6 +66,9 @@ func Test_handleFinalizer(t *testing.T) {
 						Finalizers:        []string{datadogGenericResourceFinalizer},
 						DeletionTimestamp: &metaNow,
 					},
+					Spec: datadoghqv1alpha1.DatadogGenericResourceSpec{
```

Comment:
This is necessary as the switch case for deletion handler panics on unsupported/empty resource type